metagraph/src/graph/alignment/aligner_alignment.hpp

```diff
@@ -244,8 +236,8 @@ class QueryAlignment {
     const std::vector<Alignment>& data() const { return alignments_; }
 
   private:
-    std::shared_ptr<const std::string> query_;
-    std::shared_ptr<const std::string> query_rc_;
+    std::string query_;
+    std::string query_rc_;
```

Comment:
Doing this leads to a lot of issues. Firstly, one has to either disable the copy constructor of AlignmentResults, or do some complicated pointer re-assignment when a copy happens.

Even when this is done, string optimizations in the compiler tend to screw things up...

Reply:
maybe at least change to std::unique_ptr?

Reply:
If it's a unique_ptr, then this object will no longer be copyable. The extra overhead shouldn't be an issue.

Reply:
but you don't copy it (and you don't need to), you copy a shared pointer to it.

Reply:
I ended up finding a way to get it to work with `std::string` which involved reserving the string to be just big enough to disable short-string optimizations.